geopandas/base.py

```diff
@@ -551,11 +561,21 @@ def interpolate(self, distance, normalized=False):
         Parameters
         ----------
         distance : float or Series of floats
-            Distance(s) along the geometries at which a point should be returned
+            Distance(s) along the geometries at which a point should be returned.
+            If np.array or pd.Series are used then it must have same length
+            as the GeoSeries.
         normalized : boolean
             If normalized is True, distance will be interpreted as a fraction
             of the geometric object's length.
         """
+        if isinstance(distance, (np.ndarray, pd.Series)):
+            if distance.shape[0] != self.shape[0]:
+                raise ValueError("Length of distance sequence does not match "
+                                 "length of the GeoSeries")
+            return gpd.GeoSeries(
+                    [s.interpolate(dist, normalized)
```

Comment:
Let's explicitly write `normalized=normalized` (and below, as well).

Reply:
Sure!